src/communication/xcWebSocketSubscriber.ts

```diff
@@ -15,10 +16,10 @@ export interface Subscriber {
     privateTopics: Array<String>;
     replyPublisher: Publisher;
     getHeartbeatTimer(heartbeatIntervalSeconds: number): NodeJS.Timer;
-    getModel(xcApiName: string, getModelListener: (compositionModel: CompositionModel) => void): void;
-    getXcApiList(getXcApiListListener: (apis: Array<String>) => void): void;
-    getXcApi(xcApiFileName: string, getXcApiListener: (xcApi: string) => void): void;
-    getSnapshot(componentName: string, stateMachineName: string, getSnapshotListener: (items: Array<Object>) => void): void;
+    getCompositionModel(xcApiName: string): Promise<CompositionModel>;
+    getXcApiList(): Promise<Array<String>>;
+    getXcApi(xcApiFileName: string): Promise<string>;
+    getSnapshot(componentName: string, stateMachineName: string): Promise<Array<Packet>>;
```

Comment:
Find an other name about Packet
Why not just SnapshotItem ? so get snapshot could return a Promise<Array<SnapshotItem>>...
@maurelio1234 @mbarekh what do you think about this naming?

Reply:
I thought 'Packet' was an old thing that's why i didn't comment on that. 
Packet means nothing, SnapshotItem means something :) I vote SnapshotItem

Reply:
SnapshotItem is not fine because the type packet is also used when receiving updates.
I vote StateMachineInstance